### PR TITLE
Add Support for Solaris 11

### PR DESCRIPTION
**Pull request by xanddah**
N.B. only tested on sparc-arch, should work on x86_64 as well.

### CODE REVIEW

File: manifests/client.pp [hunk 1]
```diff
@@ -140,6 +142,7 @@
 
     package { 'SYMCnbclt':
       ensure    => 'installed',
+      provider => 'sun',
```
Comment (Phil-Friderici) [inline manifests/client.pp]: Please fix the indenting here and in below cases.

File: manifests/client.pp [hunk 2]
```diff
@@ -109,6 +109,8 @@
 
   # Solaris specifc workaround
   # $my_client_packages is needed on Solaris for dependencies checks only not for package selection.
+  # Package provider sun is used since building Solaris11 packages is cumbersome
+  # and outside the scope
```
Comment (Phil-Friderici) [inline manifests/client.pp]: I guess a one liner would look better here.

**Comment by Phil-Friderici** [comment]
@xanddah Sorry for the delay, I am starting to take over the mainenance for this module now.
Firstly: thanks for your work :)

The readme should get updated to include the new supported Solaris flavour.
And finally we need spec tests for the added functionality.

**Comment by xanddah** [comment]
Ok, and thank you for your feedback!

Med vänlig hälsning/Best regards


Anders Dahlqvist

**Comment by xanddah** [comment]
Hello, Phil!
I have pushed my changes and I hope there is some magic that makes them visible to you.
Since I just as Winnie the Pooh have a very small brain this spectest thingy is pure rockescience to me  which leaves spec/classes/client_spec.rb  useless (but somewhat improved). The problem I have is with line 285 . Can I do anything of the below
a) use an array of kernelversions
b) OR them in any way
c) treat the string as a float and use a comparision , for example  => '>5.8'

**Comment by Phil-Friderici** [comment]
Hej @xanddah,

too bad the magic didn't work and I can't see a change here.
When it comes to spec tests, I can help you, no worries!

have phun
Phil

**Comment by xanddah** [comment]
Oh, I rather think it was the push that failed, not the magic. My mistake.
I have re-pushed it and now the magic sems to work.
I'll do my homework re spec-tests and come back to you if I needed.

**Comment by Phil-Friderici** [comment]
Travis is failing on puppet-lint checks. Please check the messages and fix the style issues it has.

File: manifests/client.pp [hunk 3]
```diff
@@ -107,18 +107,16 @@
...
-    $my_client_packages = 'SYMCnbclt'
+	  $my_client_packages = 'SYMCnbclt'
   } else {
-    if $client_packages == undef {
-      $my_client_packages = $default_client_packages
-    } else {
-      $my_client_packages = $client_packages
-    }
+		if $client_packages == undef {
+			$my_client_packages = $default_client_packages} else {
```
Comment (Phil-Friderici) [inline manifests/client.pp]: please don't put the ```else``` at the end of the line

**Comment by Phil-Friderici** [comment]
@xanddah strange error we got from Travis here... need to investigate myself whats going wrong

**Comment by Phil-Friderici** [comment]
puppet-lint got updated to 2.0.0 lately causing Travis to fail.

I have cherry-picked and rebased your commits, please check and confirm if my work in https://github.com/Ericsson/puppet-module-netbackup/pull/20 does fit your needs.

**Comment by Phil-Friderici** [comment]
Closing this as PR https://github.com/Ericsson/puppet-module-netbackup/pull/20 got merged inbetween.
Please re-open or create a new issue/PR if something is broken or missing.

Thanks for your thoughts and time !

**Pull request closed without merging**